helper/subproc/subproc.go

```diff
@@ -0,0 +1,67 @@
+package subproc
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"time"
+)
+
+const (
+	// ExitSuccess indicates the subprocess completed successfully.
+	ExitSuccess = iota
+
+	// ExitFailure indicates the subprocess terminated unsuccessfully.
+	ExitFailure
+
+	// ExitTimeout indicates the subprocess timed out before completion.
+	ExitTimeout
+)
+
+// MainFunc is the function that runs for this sub-process.
+//
+// The return value is a process exit code.
+type MainFunc func() int
+
+// Do f if nomad was launched as, "nomad [name]". The sub-process will exit
+// without running any other part of Nomad.
+func Do(name string, f MainFunc) {
+	if len(os.Args) > 1 && os.Args[1] == name {
+		os.Exit(f())
+	}
+}
+
+// Print the given message to standard error.
+func Print(format string, args ...any) {
+	_, _ = fmt.Fprintf(os.Stderr, format+"\n", args...)
+}
+
+// Log the given output to the logger.
+func Log(output []byte, f func(msg string, args ...any)) {
+	reader := bytes.NewReader(output)
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+		f("getter", "output", line)
+	}
```

Comment:
I just learned `strings.Split(input, "\n$")` from someone's Advent of Code submission; however in the case of untrusted input `bufio.Scanner` is nice because it protects us from reading in more than 64k per iteration by default. _Probably_ overkill in the getter case but this is a general helper function (after fixing the call to `f` here with the hardcoded name).

Reply:
> reading in more than 64k per iteration by default

`output` is all in memory already anyway. `bytes.Split` should only allocate 1 slice per newline, regardless of the amount of data between newlines, so I don't think bufio.Scanner protects us from anything?

Reply:
> avoiding loading the output of the child into memory seems like an easy win. We can do that by not using CombinedOutput and setting up an output buffer instead.

Oh... yeah... that's a way better idea. Carry on. :sweat_smile: